client/src/review-components/ReviewForm.jsx

```diff
@@ -1,76 +1,145 @@
-import React from 'react';
+import React, {useState} from 'react';
 import ReactDOM from 'react-dom/client';
-import Stars from './Stars.jsx';
+import HoverStars from './HoverStars.jsx';
+import axios from 'axios';
 
 function ReviewForm(props) {
+
+  let charDetails = {
+    Size: ['A size too small', '½ a size too small', 'Perfect', '½ a size too big', 'A size too wide'],
+    Width: ['Too narrow', 'Slightly narrow', 'Perfect', 'Slightly wide', 'Too wide'],
+    Comfort: ['Uncomfortable', 'Slightly uncomfortable', 'Ok', 'Comfortable', 'Perfect'],
+    Quality: ['Poor', 'Below average', 'What I expected', 'Pretty great', 'Perfect'],
+    Length: ['Runs short', 'Runs slightly short', 'Perfect', 'Runs slightly long', 'Runs long'],
+    Fit: ['Runs tight', 'Runs slightly tight', 'Perfect', 'Runs slightly long', 'Runs loose']
+  }
+
+  let charArray = Object.keys(props.data.characteristics).map((element, index) => {
+    return [element, props.data.characteristics[element].id, charDetails[element]]
+  })
+
+  const [formData, setFormData] = useState({
+    product_id: props.data.product_id,
+    rating: 0,
+    summary: '',
+    body: '',
+    recommend: false,
+    name: '',
+    email: '',
+    photos: [],
+    characteristics: {}
+  });
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    console.log(formData);
+    axios({
+      method: 'post',
+      url: 'http://localhost:3000/reviews',
+      data: formData,
+    })
+    .then((result) => {
+      console.log(result);
+      if (result.data === 'Created') {
+        alert('Thank you for your review!');
+        props.onClose();
+      } else {
+        alert('Something went wrong. Please try again later');
+      }
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+  }
+
+  function handleUpload(e) {
+    console.log(e.target.files);
+  }
+
+  function handleChange(e) {
+    const value = e.target.value;
+
+    if (e.target.className === 'characteristics') {
+      setFormData({...formData,
+        characteristics: {...formData.characteristics,
+        [e.target.name]: Number(value)}
```

Comment:
Number(value) can return NaN, which is of type Number, make sure you're validating that it's an actual number somewhere.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

Also, since this is setting form data, which will be sent by HTTP to a server (maybe) it'll get stringyfied. So also double check your use case, maybe you don't even want to cast into a Number type here